portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -303,21 +354,47 @@ function showResults(resultElements, wantedWords) {
       return;
     }
 
-    const textElement = document.createElement('li');
-    let resultText = result.innerText.replace(/\s\s+/g, ' ').trim();
-    resultText = resultText.replace(/[\n\r]/g, ' ');
-    const searchPos = resultText.toLowerCase().indexOf(wantedWords);
-    textElement.innerText = '...' +
-        resultText.substring(searchPos - 10, searchPos).replace(/^\s+/g, '') +
-        resultText.substring(searchPos, searchPos + 20).trim() + '...';
-
+    const textElement = createResult(result, wantedWords)[1];
     textElement.onclick = function() {
       result.scrollIntoView();
     };
 
     searchResults.appendChild(textElement);
   });
 }
+
+function showOtherResults(otherResultElements, wantedWords) {
```

Comment:
Add docstrings to these new functions. In particular I'd like to see the difference between showOtherResults and showResults described _somewhere_.

Reply:
Okay sure, I'll add those in.